### ISSUE TITLE
 Enable list type multiclass inputs to `confusion_matrix_at_thresholds`

### ISSUE THREAD

**Issue by lucyleeow** [lucyleeow/scikit-learn#17]
From here:
https://github.com/scikit-learn/scikit-learn/pull/ 32909#discussion_r2667938057

> I noticed that although we specify inputs can be 'array-like' when the input is indicator matrix, list does not work (checked on main):

```python
from sklearn.metrics import average_precision_score
y_true = [0, 0, 1, 1, 2, 2]
y_scores = [
    [0.7, 0.2, 0.1],
    [0.4, 0.3, 0.3],
    [0.1, 0.8, 0.1],
    [0.2, 0.3, 0.5],
    [0.4, 0.4, 0.2],
    [0.1, 0.2, 0.7],
]
average_precision_score(y_true, y_scores)
```
Also add a common test for list type label indicator inputs.
